docker/pypi/wmagent/bin/manage-common.sh

```diff
@@ -0,0 +1,423 @@
+
+    if [[ -f $WMA_CERTS_DIR/servicecert.pem ]] && [[ -f $WMA_CERTS_DIR/servicekey.pem ]]; then
+
+        echo "$FUNCNAME: Checking Certificate lifetime:"
+        local now=$(date +%s)
+        local certEndDate=$(openssl x509 -in $WMA_CERTS_DIR/servicecert.pem -noout -enddate)
+        certEndDate=${certEndDate##*=}
+        echo "$FUNCNAME: Certificate end date: $certEndDate"
+        [[ -z $certEndDate ]] && {
+            echo "$FUNCNAME: ERROR: Failed to determine certificate end date!"; return $(false) ;}
+        certEndDate=$(date --date="$certEndDate" +%s)
+        [[ $certEndDate -le $now ]] && {
+            echo "$FUNCNAME: ERROR: Expired certificate at $WMA_CERTS_DIR/servicecert.pem!"; return $(false) ;}
+        [[ $(($certEndDate -$now)) -le $certMinLifetimeSec ]] && {
+            echo "$FUNCNAME: WARNING: The service certificate lifetime is less than certMinLifetimeHours: $certMinLifetimeHours! Please update it ASAP!" ;}
+
+        # Renew myproxy if needed:
+        echo "$FUNCNAME: Checking myproxy lifetime:"
+        local myproxyEndDate=$(openssl x509 -in $WMA_CERTS_DIR/myproxy.pem -noout -enddate)
+        myproxyEndDate=${myproxyEndDate##*=}
+        echo "$FUNCNAME: myproxy end date: $myproxyEndDate"
+        [[ -n $myproxyEndDate ]] || $myproxyCmd && $vomsproxyCmd || {
+                echo "$FUNCNAME: ERROR: Failed to renew invalid myproxy"; return $(false) ;}
+        myproxyEndDate=$(date --date="$myproxyEndDate" +%s)
+        [[ $myproxyEndDate -gt $now ]] || $myproxyCmd && $vomsproxyCmd || {
+                echo "$FUNCNAME: ERROR: Failed to renew expired myproxy"; return $(false) ;}
+
+        # Stay safe and always change the service {cert,key} and myproxy mode here:
+        sudo chmod 600 $WMA_CERTS_DIR/*
+        echo "$FUNCNAME: OK"
+    else
+        echo "$FUNCNAME: ERROR: We found no service certificate installed at $WMA_CERTS_DIR!"
+        echo "$FUNCNAME: ERROR: Please install proper cert and key files before restarting the WMAgent container!"
+        return $(false)
+    fi
+}
+
+
+_parse_wmasecrets(){
+    # Auxiliary function to provide basic parsing of the WMAgent.secrets file
+    # :param $1: path to WMAgent.secrets file
+    local errVal=0
+    local value=""
+    local secretsFile=$1
+    # All variables need to be fetched in lowercase through: ${var,,}
+    local badValuesReg="(update-me|updateme|<update-me>|<updateme>|fix-me|fixme|<fix-me>|<fixme>|^$)"
+    local varsToCheck=`awk -F\= '{print $1}' $secretsFile | grep -vE "^[[:blank:]]*#.*$"`
+    for var in $varsToCheck
+    do
+        value=`grep -E "^[[:blank:]]*$var" $secretsFile | awk -F\= '{print $2}'`
+        [[ ${value,,} =~ $badValuesReg ]] && { echo "$FUNCNAME: Bad value for: $var=$value"; let errVal+=1 ;}
+    done
+    return $errVal
+}
+
+
+#
+# Passwords/Secrets handling
+#
+_load_wmasecrets(){
+    if [ "x$WMA_SECRETS_FILE" == "x" ]; then
+        WMA_SECRETS_FILE=$HOME/WMAgent.secrets;
```

Comment:
Just double-checking whether $HOME is the right/expected location for this file?

Reply:
Well this is taken as is from the previous implementation of the `manage script`. Back then obviously the agent was checking if the `$WMA_SECRETS_FILE` env variable has been set in advance and if not  was falling back to taking the `WMAgent.secrets` from `$HOME`. I pretty much willing to remove that one.